### PR TITLE
setup ArbitraryTransformer with new testing appraoch

### PR DESCRIPTION
**Pull request by limlam96**
Issue: https://github.com/lvgig/tubular/issues/207

Made changes to bring ArbitraryImputer in line with new test format, also refactored BaseImputer tests to ease the rest of the rollout across imputers. 

Introduced new 'FITS' class attribute to simplify test inheritance process and turning on/off relevant tests, if the reviewer approves then suggest to rollout across transformers. Possibly default to True in BaseTransformer and set to False where needed in main transformers.

### CODE REVIEW

File: tests/imputers/test_ArbitraryImputer.py [hunk 1]
```diff
@@ -167,3 +77,15 @@ def test_impute_value_preserve_dtype(self):
         # Checking if the dtype of "a" and "b" are int8 and float16 respectively after imputation
         assert df["a"].dtype == "int8"
         assert df["b"].dtype == "float16"
+
+
```
Comment (lsumption) [inline tests/imputers/test_ArbitraryImputer.py:81]: Why are we not using GenericInitTests here?

**Comment by lsumption** [comment]
Should we be adding type hints to the tests?

**Comment by lsumption** [comment]
Can we update the changelog with these changes?

File: tests/imputers/test_ArbitraryImputer.py (hunk 1 above)
Comment (limlam96) [reply tests/imputers/test_ArbitraryImputer.py:81]: as in why ColumnStrListInitTests?

ColumnStrListInitTests inherits from GenericInitTests and adds checks for columns=List[str], so just adds a few more relevant tests to GenericInitTests :)

File: tests/imputers/test_BaseImputer.py [hunk 2]
```diff
@@ -99,56 +91,85 @@ def expected_df_3():
...
     )
-    def test_expected_output_3(self, df, expected):
+    def test_expected_output_3(self, df, expected, initialized_transformers):
         """Test that transform is giving the expected output when applied to object and categorical columns."""
-        x1 = BaseImputer(columns=["b", "c"])
+        x1 = initialized_transformers[self.transformer_name]
 
         x1.impute_values_ = {"b": "g", "c": "f"}
+        x1.columns = ["b", "c"]
+        # bit of a hack to make this work nicely for arbitrary imputer
```
[the remark this answers is not in the document]
Comment (limlam96) [reply tests/imputers/test_BaseImputer.py]: yeah sorry that wasn't clear haha, updated here https://github.com/lvgig/tubular/pull/212/commits/75fe43633a8fb88bff925863a32c81f37c75744d

**Comment by limlam96** [comment]
updated changelog here https://github.com/lvgig/tubular/pull/212/commits/f5234c86c0a94857a4a1508b70870e0a515fd86f

**Comment by limlam96** [comment]
re type hints in tests.. maybe? 

But think that would be one to spin out into another issue as it doesn't currently seem to be done elsewhere

**Comment by lsumption** [comment]
Happy to leave the type hints for now. I think there's just an outstanding merge conflict then happy to approve!

**Comment by limlam96** [comment]
nice thanks :) should be resolved now!

**Review by lsumption — APPROVED** [review]

**Pull request merged**